### PR TITLE
[Refactor] Refact code of sequence column

### LINKED ISSUE #7006: [Refactor] Refactor code of sequence column
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

1. refactoring code to remove part of the redundant and redundant logic
2. add sequence column as sorting rule to CollectIterator and SegmentUnion to simplify the reading logic of unique_key

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by HappenLee**
## Proposed changes

#7006 

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [x] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue on (Fix #7006) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If these changes need document changes, I have updated the document
- [x] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/exec/olap_scanner.cpp [hunk 1]
```diff
@@ -245,8 +246,9 @@ Status OlapScanner::_init_return_columns() {
                 break;
             }
         }
-        if (has_replace_col) {
-            _return_columns.push_back(_tablet->tablet_schema().sequence_col_idx());
+        if (auto sequence_col_idx = _tablet->tablet_schema().sequence_col_idx();
+            has_replace_col && std::find(_return_columns.begin(), _return_columns.end(), sequence_col_idx) == _return_columns.end()) {
+            _return_columns.push_back(sequence_col_idx);
```
Comment (wangbo) [inline be/src/exec/olap_scanner.cpp:251]: What case may cause sequence_col_idx insert to return columns duplicately

Comment (HappenLee) [reply be/src/exec/olap_scanner.cpp:251]: it's a bug, i fixed it.
if you 'set show_hidden_columns = true;', select * from unique_table with sequnece column. In Debug mode will core.

**Review by wangbo — DISMISSED** [review]
LGTM

**Review by wangbo — APPROVED** [review]
LGTM

**Pull request merged**